src/sinks/honeycomb/sink.rs

```diff
@@ -0,0 +1,72 @@
+//! Implementation of the `honeycomb` sink.
+
+use crate::sinks::{prelude::*, util::http::HttpRequest};
+
+use super::request_builder::HoneycombRequestBuilder;
+
+pub(super) struct HoneycombSink<S> {
+    service: S,
+    batch_settings: BatcherSettings,
+    request_builder: HoneycombRequestBuilder,
+}
+
+impl<S> HoneycombSink<S>
+where
+    S: Service<HttpRequest> + Send + 'static,
+    S::Future: Send + 'static,
+    S::Response: DriverResponse + Send + 'static,
+    S::Error: std::fmt::Debug + Into<crate::Error> + Send,
+{
+    /// Creates a new `HoneycombSink`.
+    pub(super) const fn new(
+        service: S,
+        batch_settings: BatcherSettings,
+        request_builder: HoneycombRequestBuilder,
+    ) -> Self {
+        Self {
+            service,
+            batch_settings,
+            request_builder,
+        }
+    }
+
+    async fn run_inner(self: Box<Self>, input: BoxStream<'_, Event>) -> Result<(), ()> {
+        let service = ServiceBuilder::new().service(self.service);
+        input
+            // Batch the input stream with byte size calculation
+            .batched(self.batch_settings.into_byte_size_config())
```

Comment:
💭 Similar to the http sink, should we batch based off the estimated json size instead? It seems to be more suitable given the context?